selfdrive/car/subaru/carcontroller.py

```diff
@@ -1,32 +1,34 @@
-#from common.numpy_fast import clip
 from selfdrive.car import apply_std_steer_torque_limits
 from selfdrive.car.subaru import subarucan
-from selfdrive.car.subaru.values import DBC
+from selfdrive.car.subaru.values import DBC, CAR
 from opendbc.can.packer import CANPacker
 
 
 class CarControllerParams():
-  def __init__(self):
-    self.STEER_MAX = 2047              # max_steer 4095
-    self.STEER_STEP = 2                # how often we update the steer cmd
-    self.STEER_DELTA_UP = 50           # torque increase per refresh, 0.8s to max
-    self.STEER_DELTA_DOWN = 70         # torque decrease per refresh
-    self.STEER_DRIVER_ALLOWANCE = 60   # allowed driver torque before start limiting
-    self.STEER_DRIVER_MULTIPLIER = 10  # weight driver torque heavily
-    self.STEER_DRIVER_FACTOR = 1       # from dbc
-
+  def __init__(self, car_fingerprint):
+    self.STEER_MAX = 2047                # max_steer 2047
+    self.STEER_STEP = 2                  # how often we update the steer cmd
+    self.STEER_DELTA_UP = 50             # torque increase per refresh, 0.8s to max
+    self.STEER_DELTA_DOWN = 70           # torque decrease per refresh
+    if car_fingerprint in [CAR.ASCENT, CAR.FORESTER, CAR.IMPREZA]:
+      self.STEER_DRIVER_ALLOWANCE = 60   # allowed driver torque before start limiting
+    if car_fingerprint == CAR.LEGACY_2015:
```

Comment:
same thing about maintaining a list of pre-global cars

Reply:
done